microsoft/fluentui-apple
# Enable Fluent-style back button for large/leading titles

### Platforms Impacted

### Description of changes

This allows a back button to be shown on navigation bars with large/leading titles. We also ensure that we use the Fluent-style back button (which has a thinner stroke than vanilla UIKit's back button) and ensure that leading spacing is correct.

### Verification

Testing has been done in the following areas:
* Ensure that we can navigate between a root view controller with both leading and centered titles.
* Ensure that the back button text is labeled appropriately when navigating from either direction (i.e., from root to child and from grandchild to child).
* Ensure that the back button faces the correct direction in RTL languages.
* Ensure that the layout is correct on iPhones running iOS 14, iOS 15, and iOS 16. (As of 3fe3ed738403b889244e26fc6781d08af71fef59, there's a known issue where the back button on centered titles is misaligned when we have a full-width app on iPad. We're still investigating this.)

| Leading | Centered | Leading (RTL) | Centered (RTL) |
|-|-|-|-|
| ![Simulator Screen Shot - iPhone 14 Pro - 2023-02-15 at 11 44 55](https://user-images.githubusercontent.com/717674/219136430-bf4d6ee9-a791-41ba-90b7-dd9329091547.png) | ![Simulator Screen Shot - iPhone 14 Pro - 2023-02-15 at 11 45 04](https://user-images.githubusercontent.com/717674/219136457-b3e6798f-c40f-4fdf-91b4-4074278f104b.png) | ![Simulator Screen Shot - iPhone 14 Pro - 2023-02-15 at 11 40 59](https://user-images.githubusercontent.com/717674/219136629-cdc16983-b876-4c2d-be2a-a96fe1e52ded.png) | ![Simulator Screen Shot - iPhone 14 Pro - 2023-02-15 at 11 41 03](https://user-images.githubusercontent.com/717674/219136645-f5e96b88-4012-4269-bfc1-27f4c1d1d9f5.png) |

### Test App Changes

* Added a new test case where we show a custom leading button in the root view instead of the avatar.
* The disclosure button in child cell 6 now changes the style of the title bar. This is more visible in a standalone app than `NSLog`. (I would've shown a `UIAlert`, but since the navigation controller we're demoing is already detached, it's considered bad practice to show another modal dialog.)

| Custom leading button | Disclosure action |
|-|-|
| <img src="https://user-images.githubusercontent.com/717674/219155992-e5d48102-db7a-4189-999d-323953af5d50.png" width="295"> | ![Simulator Screen Recording - iPhone 14 Pro - 2023-02-15 at 13 33 14](https://user-images.githubusercontent.com/717674/219173150-b187b575-85dc-42bc-bdf2-8f4883dd6b89.gif) |



### Pull request checklist

This PR has considered:
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)
